src/components/HeaderSearch/index.tsx

```diff
@@ -7,29 +7,33 @@ import Bind from 'lodash-decorators/bind'
 const styles = require('./index.less')
 
 interface HeaderSearchProps {
-  className: string
-  placeholder: string
-  open: boolean
-  onChange: Function
-  onPressEnter: Function
-  defaultOpen: boolean
+  placeholder?: string
+  dataSource?: string[]
+  defaultOpen?: boolean
+  open?: boolean
+  onSearch?: (value: string) => void
+  onChange?: (value: string) => void
+  onVisibleChange?: (visible: boolean) => void
+  onPressEnter?: (value: string) => void
+  className?: string
 }
 
 export default function HeaderSearch (props: HeaderSearchProps) {
   const [searchMode, setSearchMode] = useState(props.defaultOpen)
   const [value, setValue] = useState('')
   // TO Verify: onVisibleChange not defined or passed as props
   const { className, placeholder, open, onChange, onPressEnter, ...restProps } = props
-  let textInput = React.createRef()
+  const textInput = createRef()
 
   let timeout
```

Comment:
what's the value of the `timeout`?

Reply:
yes, it is used to hold value. Not necessary, removed